environment/src/main/java/com/sequenceiq/environment/proxy/repository/ProxyConfigRepository.java

```diff
@@ -0,0 +1,28 @@
+package com.sequenceiq.environment.proxy.repository;
+
+import java.util.Optional;
+import java.util.Set;
+
+import javax.transaction.Transactional;
+import javax.transaction.Transactional.TxType;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+import com.sequenceiq.environment.proxy.domain.ProxyConfig;
+
+@Transactional(TxType.REQUIRED)
+public interface ProxyConfigRepository extends JpaRepository<ProxyConfig, Long> {
+
+    @Query("SELECT p FROM ProxyConfig p WHERE p.accountId = :accountId")
+    Set<ProxyConfig> findAllInAccount(@Param("accountId") String accountId);
+
+    @Query("SELECT p FROM ProxyConfig p WHERE p.accountId= :accountId AND (p.name= :name OR p.resourceCrn= :name) "
+            + "AND p.archived IS FALSE")
+    Optional<ProxyConfig> findByNameInAccountId(@Param("name") String name, @Param("accountId") String accountId);
```

Comment:
Method name is a little short-spoken regarding the amount of filtering what the query does.

Reply:
findNotArchivedByAccountIdAndNameOrCrn

Reply:
ok